test/functional/lua/buffer_updates_spec.lua

```diff
@@ -420,6 +420,35 @@ describe('lua buffer event callbacks: on_lines', function()
     feed('i<Tab>')
     eq({ '\ta' }, exec_lua('return _G.res[#_G.res]'))
   end)
+
+  it('quickfix buffer send change', function()
+    command('copen')
+    exec_lua(function()
+      vim.api.nvim_buf_attach(vim.api.nvim_get_current_buf(), false, {
+        on_lines = function(...)
+          vim.g.qf_on_lines = { ... }
+        end,
+        on_bytes = function(...)
+          vim.g.qf_on_bytes = { ... }
```

Comment:
The other tests use `check_events`. Using a different approach makes the tests harder to understand. OTOH, this test is testing both "lines" and "bytes".